src/engine/runtime.js

```diff
@@ -435,12 +435,24 @@ Runtime.prototype.startHats = function (requestedHatOpcode,
             // Not the right hat.
             return;
         }
+
         // Match any requested fields.
         // For example: ensures that broadcasts match.
         // This needs to happen before the block is evaluated
         // (i.e., before the predicate can be run) because "broadcast and wait"
         // needs to have a precise collection of started threads.
         var hatFields = target.blocks.getFields(topBlockId);
+
+        // If no fields are present, check inputs (horizontal blocks)
+        if (Object.keys(hatFields).length === 0) {
+            var hatInputs = target.blocks.getInputs(topBlockId);
+            for (var input in hatInputs) {
```

Comment:
I would suggest putting something like this at the top of the loop body:
```js
if (!hatInputs.hasOwnProperty(input)) continue;
```

We're actually missing this sort of check in several places. If you'd like to see the VM explode, try typing something like this into your JS console while the VM is running:
```js
Object.prototype.harmless = false;
```

I'm of the opinion that we should try to fix this, but I'd be happy to discuss if you disagree. Also, even if you agree that we should fix this sort of thing, since it's an issue in several places I don't feel the need to hold this bug-fix up. We can do a `hasOwnProperty` pass at some other time.